File: js_code/135_Candy.js
/**
 * @param {number[]} ratings
 * @return {number}
 */
 var candy = function(ratings) {
    let sum=0,length=ratings.length;
    let ans = new Array(length).fill(1);//简便初始化方式
    // for(let i=0; i<ratings.length; i++)
    // {ans.push(1);}
    for(let i=1;i<length; i++)
    {
      if(ratings[i]>ratings[i-1])
      {
        ans[i]=ans[i-1]+1;
      }
    }
    for(let i=length-1;i>=1;i--)
    {
      if(ratings[i-1]>ratings[i])
      {
        ans[i-1]=Math.max(ans[i]+1,ans[i-1]);
      }
    }
    for(let i=0;i<ans.length; i++)
    {
      sum+=ans[i];
      console.log(ans[i])
    }
    console.log(sum);
    return sum;
};

candy([1,0,2])

File: py_code/135_Candy.py
from typing import List

class Solution:
    def candy(self, ratings: List[int]) -> int:
      ans=[1]*len(ratings) ##初始化的方式
      sum=0
      # for x in ratings:
      #   ans.append(1)
      for i in range(1,len(ratings)):
        if ratings[i]>ratings[i-1]:
          ans[i]=ans[i-1]+1
      for i in range(len(ratings)-1,0,-1):
        if ratings[i-1]>ratings[i]:
          ans[i-1]=max(ans[i]+1,ans[i-1])        
      for x in ans:
        print(x)
        sum+=x
      print(sum)
      return sum;

s=Solution()
s.candy([1,0,2])